java/src/main/java/com/globalpayments/example/ProcessPaymentServlet.java

```diff
@@ -171,346 +190,87 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
 
             String currency = request.getParameter("currency").trim().toUpperCase();
 
-            // Create payment link using direct API call (matching Node.js approach)
-            String paymentLinkResponse = createPaymentLinkViaAPI(amount, currency, reference, name, description);
+            // Create payment link using SDK (matching PHP approach)
+            String paymentLinkResponse = createPaymentLinkViaSdk(amount, currency, reference, name, description);
 
             response.getWriter().write(paymentLinkResponse);
 
+        } catch (ApiException e) {
+            // Handle API-specific errors
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            String errorResponse = String.format(
+                "{\"success\":false,\"message\":\"Payment link creation failed\",\"error\":{\"code\":\"API_ERROR\",\"details\":\"%s\"}}",
+                e.getMessage().replace("\"", "\\\"")
+            );
+            response.getWriter().write(errorResponse);
         } catch (Exception e) {
-            // Handle payment link creation errors
+            // Handle other errors
             response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
             String errorResponse = String.format(
-                "{\"success\":false,\"message\":\"Payment link creation failed\",\"error\":{\"code\":\"API_ERROR\",\"details\":\"%s\"}}",
+                "{\"success\":false,\"message\":\"Payment link creation failed\",\"error\":{\"code\":\"INTERNAL_ERROR\",\"details\":\"%s\"}}",
                 e.getMessage().replace("\"", "\\\"")
             );
             response.getWriter().write(errorResponse);
         }
     }
 
     /**
-     * Creates a payment link using direct GP API HTTP calls.
-     * Matches the Node.js implementation approach with access token generation.
+     * Creates a payment link using the Global Payments SDK.
+     * Matches the PHP implementation approach using PayByLinkService.
```

Comment:
Only nitpick, but would just remove any reference of other implementations in the code. I thought I cleaned it out but seems like I kept it in 🤔